pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -176,8 +176,18 @@ func calcRestartCountByLogDir(path string) (int, error) {
 func (m *kubeGenericRuntimeManager) startContainer(ctx context.Context, podSandboxID string, podSandboxConfig *runtimeapi.PodSandboxConfig, spec *startSpec, pod *v1.Pod, podStatus *kubecontainer.PodStatus, pullSecrets []v1.Secret, podIP string, podIPs []string) (string, error) {
 	container := spec.container
 
+	// Get the runtime handler so that we can pass it to image puller.
+	runtimeHandler := ""
+	if m.runtimeClassManager != nil {
+		var err error
+		runtimeHandler, err = m.runtimeClassManager.LookupRuntimeHandler(pod.Spec.RuntimeClassName)
+		if err != nil {
+			return fmt.Sprintf("Failed to get runtime class for pod %q: %v", format.Pod(pod), err), err
+		}
+	}
+
 	// Step 1: pull the image.
-	imageRef, msg, err := m.imagePuller.EnsureImageExists(ctx, pod, container, pullSecrets, podSandboxConfig)
+	imageRef, msg, err := m.imagePuller.EnsureImageExists(ctx, pod, runtimeHandler, container, pullSecrets, podSandboxConfig)
```

Comment:
> Maybe the best route is just to add the handler as an annotation in the pod spec, then the copy to the image services would be automatic by pod annotation copy.

Yes, I think this would simplify this patch. Let me take a stab at it.

Reply:
@mikebrow I just uploaded a version that I believe does what you suggest and is much simpler, just the addition of 7 lines overall, including test.

Please let me know what you think!